### PR TITLE
Adding core changes for future Sonar Webhook into API

### PR DESCRIPTION
**Pull request by shriver135**

### CODE REVIEW

File: src/main/java/com/capitalone/dashboard/model/SonarProject.java [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+        getOptions().put(PROJECT_NAME, name);
+    }
+
+    @Override
+    public boolean equals(Object o) {
+        if (this == o) return true;
+        if (o == null || getClass() != o.getClass()) return false;
+
+        SonarProject that = (SonarProject) o;
+        return getProjectId().equals(that.getProjectId()) && getInstanceUrl().equals(that.getInstanceUrl());
```
Comment (rvema) [inline src/main/java/com/capitalone/dashboard/model/SonarProject.java]: add a null check for `that` as well

File: src/main/java/com/capitalone/dashboard/model/SonarProject.java [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+        getOptions().put(PROJECT_NAME, name);
+    }
+
+    @Override
+    public boolean equals(Object o) {
+        if (this == o) return true;
+        if (o == null || getClass() != o.getClass()) return false;
+
+        SonarProject that = (SonarProject) o;
+        if (that == null) return false;
```
Comment (shriver135) [inline src/main/java/com/capitalone/dashboard/model/SonarProject.java]: Added NullCheck

Comment (rvema) [reply src/main/java/com/capitalone/dashboard/model/SonarProject.java]: please change `if (o == null || getClass() != o.getClass()) return false;` to `if (o == null || !(this instanceof o)) return false;`

Comment (shriver135) [reply src/main/java/com/capitalone/dashboard/model/SonarProject.java]: This should be resolved.

**Pull request merged**